packages/lesswrong/server/recommendations.js

```diff
@@ -0,0 +1,140 @@
+import { addGraphQLResolvers, addGraphQLQuery } from 'meteor/vulcan:core';
+import { Posts } from '../lib/collections/posts';
+import { WeightedList } from './weightedList.js';
+
+const scoreRelevantFields = {baseScore:1, curatedDate:1, frontpageDate:1};
+
+const pipelineFilterUnread = ({currentUser}) => {
+  if (!currentUser)
+    return [];
+  
+  return [
+    { $lookup: {
+      from: "lwevents",
+      let: { documentId: "$_id", },
+      pipeline: [
+        { $match: {
+          name: "post-view",
+          userId: currentUser._id,
+        } },
+        { $match: { $expr: {
+          $and: [
+            {$eq: ["$documentId", "$$documentId"]},
+          ]
+        } } },
+        { $limit: 1},
+      ],
+      as: "views",
+    } },
+    
+    { $match: {
+      "views": {$size:0}
+    } },
+  ];
+}
+
+const defaultViewPostFilter = {
```

Comment:
Use a `getParameters` call here to actually get the defaultView filters instead of duplicating them